packages/java/endpoint/src/main/java/dev/hilla/EndpointInvoker.java

```diff
@@ -384,6 +385,12 @@ private Object invokeVaadinEndpointMethod(String endpointName,
         var invokedEndpointClass = vaadinEndpointData.getEndpointObject()
                 .getClass();
 
+        // if the invoked endpoint is proxied, then take the original class:
+        // This is only to supports CGLIB proxies as Spring AOP proxies are.
+        if (Enhancer.isEnhanced(invokedEndpointClass)) {
+            invokedEndpointClass = invokedEndpointClass.getSuperclass();
+        }
```

Comment:
If there's no way to avoid coupling with Spring, I would suggest using `ClassUtils.getUserClass`, as already done below in this method

```java
        Type[] javaParameters = getJavaParameters(methodToInvoke, ClassUtils
                .getUserClass(vaadinEndpointData.getEndpointObject()));
```